aws_gcp_vpn/main.tf

```diff
@@ -0,0 +1,163 @@
+      preshared_key           = aws_vpn_connection.default["0"].tunnel1_preshared_key,
+    },
+    1 = {
+      aws_external_ip_address = aws_vpn_connection.default["0"].tunnel2_address,
+      aws_internal_ip_address = aws_vpn_connection.default["0"].tunnel2_vgw_inside_address
+      gcp_gateway_id          = 0,
+      gcp_internal_ip_address = aws_vpn_connection.default["0"].tunnel2_cgw_inside_address
+      internal_ip_range       = aws_vpn_connection.default["0"].tunnel2_inside_cidr
+      preshared_key           = aws_vpn_connection.default["0"].tunnel2_preshared_key,
+    },
+    2 = {
+      aws_external_ip_address = aws_vpn_connection.default["1"].tunnel1_address,
+      aws_internal_ip_address = aws_vpn_connection.default["1"].tunnel1_vgw_inside_address
+      gcp_gateway_id          = 1,
+      gcp_internal_ip_address = aws_vpn_connection.default["1"].tunnel1_cgw_inside_address
+      internal_ip_range       = aws_vpn_connection.default["1"].tunnel1_inside_cidr
+      preshared_key           = aws_vpn_connection.default["1"].tunnel1_preshared_key,
+    },
+    3 = {
+      aws_external_ip_address = aws_vpn_connection.default["1"].tunnel2_address,
+      aws_internal_ip_address = aws_vpn_connection.default["1"].tunnel2_vgw_inside_address
+      gcp_gateway_id          = 1,
+      gcp_internal_ip_address = aws_vpn_connection.default["1"].tunnel2_cgw_inside_address
+      internal_ip_range       = aws_vpn_connection.default["1"].tunnel2_inside_cidr
+      preshared_key           = aws_vpn_connection.default["1"].tunnel2_preshared_key,
+    },
+  }
+  gcp_external_gateway_ips = {
+    0 = google_compute_ha_vpn_gateway.default.vpn_interfaces[0],
+    1 = google_compute_ha_vpn_gateway.default.vpn_interfaces[1],
+  }
+  # List of supported ciphers
+  # https://cloud.google.com/network-connectivity/docs/vpn/concepts/supported-ike-ciphers#ikev2_ciphers_that_use_aead
+  # Note that GCP requires AWS to advertise fewer configuration options for the VPN rekeying to work
+  # https://cloud.google.com/network-connectivity/docs/vpn/how-to/creating-ha-vpn#create_ha_vpn_to_aws_peer_gateways
+  vpn_dh_group_numbers      = [21] # ecp521
+  vpn_encryption_algorithms = ["AES256-GCM-16"]
+  vpn_ike_versions          = ["ikev2"]
+  vpn_integrity_algorithms  = ["SHA2-512"]
+}
+
+resource "google_compute_router" "default" {
+  name    = "gcp-to-aws-cloud-router"
+  network = var.gcp_network_name
+
+  description = "GCP to AWS Cloud Router"
+
+  bgp {
+    asn = var.gcp_private_asn
+  }
+}
+
+resource "google_compute_ha_vpn_gateway" "default" {
+  name    = "gcp-to-aws-vpn-gateway"
+  network = var.gcp_network_name
+}
+
+resource "aws_customer_gateway" "default" {
+  for_each = local.gcp_external_gateway_ips
+
+  bgp_asn    = var.gcp_private_asn
+  ip_address = each.value.ip_address
+  type       = "ipsec.1"
+
+  tags = {
+    Name = "gcp-${var.gcp_project_id}-to-aws-cg${each.value.id}"
+  }
+}
+
+# FIXME This only needs to be created once per AWS account
```

Comment:
If it's once per account then maybe outside of this module and add it as a documentation requirement? We just need to aws_vpn_gateway.id as an input.

Reply:
mmm yeah

Reply:
Or maybe just a resource that can be flagged on or off.